tests/FSharp.Test.Utilities/ProjectGeneration.fs

```diff
@@ -896,4 +1176,51 @@ type SyntheticProject with
     /// Saves project to disk and checks it with default options. Returns the FSharpChecker that was created
     member this.SaveAndCheck() =
         this.Workflow.Yield() |> Async.RunSynchronously |> ignore
-        this.Workflow.Checker
\ No newline at end of file
+        this.Workflow.Checker
+
+    static member CreateFromRealProject(projectDir) =
+
+        let projectFile =
+            projectDir
+            |> Directory.GetFiles
+            |> Seq.filter (fun f -> f.EndsWith ".fsproj")
+            |> Seq.toList
+            |> function
+                | [] -> failwith $"No .fsproj file found in {projectDir}"
+                | [ x ] -> x
+                | files -> failwith $"Multiple .fsproj files found in {projectDir}: {files}"
+
+        let fsproj = XmlDocument()
+        do fsproj.Load projectFile
+
+        let signatureFiles, sourceFiles =
```

Comment:
There is this little voice telling me "nobody sane should attempt to parse .fsproj and think it will be easy".
Btw. did you check https://github.com/ionide/proj-info ?

Reply:
I just wanted something simple. Because it's not only parsing that's required but also modelling everything with the `SyntheticProject` type. But maybe I'll attempt it at some point for FCS, which would be nice to have for testing and benchmarking. Then I'd use proj-info to replace this stuff.